### ISSUE TITLE
Context Encoder implementation flaws

### ISSUE THREAD

**Issue by marcelogdeandrade** [eriklindernoren/Keras-GAN#44]
Hello, it seems that the context encoder implementation has some flaws compared to the paper described: https://arxiv.org/abs/1611.06430

First of all, the optimizer used for both generator and discriminator are the same. On the paper is described that the generator and discriminator have a different learning rate, it being 10 times bigger on the generator.

Second, one of the main points on the generator network is the fully connected layer on the bottleneck, which is missing on the implementation network aswell.

Thanks
